source/layers/validation/checkers/system_resource_tracker/system_resource_tracker.md

```diff
@@ -0,0 +1,382 @@
+# System Resource Tracker Checker
+
+## Overview
+
+The System Resource Tracker is a Level Zero validation layer checker that monitors both Level Zero API resources and system resources in real-time. It tracks resource allocation and deallocation across all Level Zero API calls that create or destroy resources, providing detailed insights into memory usage, resource lifecycles, and system-level metrics.
+
+**Platform Support:** Linux only. This checker uses `/proc/self/status` for system metrics and is not available on Windows or macOS.
+
+## Features
+
+- **Level Zero Resource Tracking**: Monitors all L0 resources including contexts, command queues, modules, kernels, event pools, command lists, events, fences, images, samplers, and memory allocations
+- **Memory Leak Detection**: Automatically detects memory leaks by tracking handle-to-metrics mappings for all create/destroy operations:
+  - Compares memory usage at resource creation vs destruction
+  - Logs warnings when memory increases during destroy operations
+  - Reports cumulative leaks per resource type at program exit
+  - Provides detailed per-handle leak information
+- **System Resource Monitoring**: Tracks real system metrics via `/proc/self/status` including:
+  - Virtual memory size (VmSize)
+  - Resident set size (VmRSS)
+  - Data segment size (VmData)
+  - Peak virtual memory (VmPeak)
+  - Thread count
+  - File descriptor count
+- **Signed Delta Tracking**: Calculates both positive and negative resource changes (deltas) for each API call with proper signed arithmetic
+- **Cumulative Summaries**: Maintains running totals of all resource types and leak totals
+- **CSV Export**: Optionally exports timestamped data for graphing and analysis
+- **Per-Process Isolation**: Each process creates a unique output file with PID appended
+- **Thread-Safe**: Uses mutex protection for concurrent access using Meyers' singleton pattern
+- **Low Overhead**: Minimal performance impact (< 1ms per API call)
+
+## Usage
+
+### Basic Tracking with Debug Logging
+
+Enable the checker to log resource usage to the Level Zero debug log:
+
+```bash
+export ZE_ENABLE_VALIDATION_LAYER=1
+export ZEL_ENABLE_SYSTEM_RESOURCE_TRACKER_CHECKER=1
+export ZEL_ENABLE_LOADER_LOGGING=1
+export ZEL_LOADER_LOGGING_LEVEL=debug
+
+# Run your Level Zero application
+./my_level_zero_app
+```
+
+### CSV Output for Graphing
+
+Set the `ZEL_SYSTEM_RESOURCE_TRACKER_CSV` environment variable to specify the output CSV file path:
+
+```bash
+export ZE_ENABLE_VALIDATION_LAYER=1
+export ZEL_ENABLE_SYSTEM_RESOURCE_TRACKER_CHECKER=1
+export ZEL_SYSTEM_RESOURCE_TRACKER_CSV=tracker_output.csv
```

Comment:
Not a change request, just a question, which path is this assuming as it's default... 
The CWD of the application, or the CWD of the L0 Loader

Reply:
CWD of the application